File: py/g.py
import itertools as it
import e, aoc   #e.py; day 5 code
s = aoc.sread('i.7', int, ',')
for r in (range(5), range(5,10)):   #both parts' code will terminate correctly
    final = []
    for t in it.permutations(r):
        gens, ls = [e.phaseinit(s, t[i])[0] for i in range(5)], []
        send, cont = 0, True
        while cont:
            for i in range(5):
                send = gens[i].send(send)
                try: gens[i].next()
                except Exception: cont = False
            ls.append(send) #for part 1, len(ls) == 1
        final.append(max(ls))
    print max(final)    #print answers

File: incomplete/g.js
const aoc = require("./aoc");
const e = require("./e");
const s = aoc.sread("i.7", Number, ',');
//part 2 not functional
[aoc.range(5), aoc.range(5, 10)].forEach(r => {
    const fin = [];
    for (let t of aoc.permute(r)) {
        const gens = Array.from(aoc.range(5), i => e.phaseInit(s, t[i])[0]);
        const ls = [];
        let cont = true;
        while (cont) {
            let send = 0;
            for (let i of aoc.range(5)) {
                send = gens[i].next(send).value;
                if (gens[i].next().done)
                    cont = false;
            }
            //if (send !== undefined)
            ls.push(send);
        }
        fin.push(ls.reduce(aoc.max));
    }
    console.log(fin.reduce(aoc.max));
});
